components/templates/ListOfTests.tsx

```diff
@@ -0,0 +1,145 @@
+    },
+    container: {
+        paddingTop: theme.spacing(4),
+        maxWidth: 3400,
+        paddingBottom: theme.spacing(4)
+    },
+    paper: {
+        padding: theme.spacing(2),
+        display: "flex",
+        overflow: "auto",
+        flexDirection: "column"
+    },
+    padding: {
+        marginBottom: "5px",
+        marginLeft: "80%",
+        color: "#353690"
+    },
+    nameOfTestOrSuiteDark: {
+        paddingLeft: theme.spacing(4),
+        fontSize: "0.875rem",
+        textAlign: "left",
+        fontFamily: "Roboto",
+        fontWeight: 400,
+        wordBreak: "break-all",
+        whiteSpace: "pre-wrap",
+        color: theme.palette.secondary.light
+    },
+    nameOfTestOrSuiteLight: {
+        paddingLeft: theme.spacing(4),
+        fontSize: "0.875rem",
+        textAlign: "left",
+        fontFamily: "Roboto",
+        fontWeight: 400,
+        wordBreak: "break-all",
+        whiteSpace: "pre-wrap"
+    },
+    backgroundGrey: {
+        backgroundColor: "#d6d6d6"
+    },
+    backgroundDarkGrey: {
+        backgroundColor: "#474747"
+    },
+    backgroundWhite: {
+        backgroundColor: "white"
+    },
+    backgroundDark: {
+        backgroundColor: theme.palette.secondary.main,
+        border: "1px grey solid"
+    },
+    testBackgroundDark: {
+        backgroundColor: theme.palette.secondary.main
+    }
+}))
+
+
+type Props = {
+  showTest: Function
+  highlightedTest: number
+  children: Test[]
+  darkMode: boolean
+}
+
+export const ListOfTests = function (props : Props) {
+const {showTest, highlightedTest, children, darkMode} = props
+const classes = useStyles(props)
+const [tests, setTests] = useState(children || [])
+
+function setTextLineStyle(isHighlighted, darkMode) {
+    if (isHighlighted && ! darkMode)  // light mode - highlighted
+        return classes.backgroundGrey
+     else if (isHighlighted && darkMode)  // dark mode - highlighted
+        return classes.backgroundDarkGrey
+     else if (isHighlighted && ! darkMode)  // light mode - not highlighted
+        return classes.backgroundWhite
+     else if (! isHighlighted && darkMode)  // dark mode - not highlighted
+        return classes.testBackgroundDark
+}
+
+const updateTest = (index, test) => {
+  const newTests = [...tests];
+  newTests[index] = test;
+  setTests(newTests);
+}
+
+useSocket('delta_resolution', testResolution => {
+  let filteredTest = tests.find(
+    test => test.test_id === testResolution.test_id);
+    if (filteredTest){
+      let testIndex = tests.indexOf(filteredTest);
+      if (filteredTest.test_history_id == testResolution.test_history_id){
+        filteredTest.test_history_resolution = testResolution.test_history_resolution
+        updateTest(testIndex, filteredTest)
+      } else if (filteredTest.test_history_resolution == 1 || !filteredTest.test_history_resolution){
+        filteredTest.test_resolution = testResolution.test_resolution
+        updateTest(testIndex, filteredTest)
```

Comment:
logic in this function seems kinda complicated, maybe we can simplify it later. Do you mind putting a small comment explaining those ifs, pleeeease? 🙏

Reply:
Cool comments just added 👍

Reply:
still don't fully understand what those ifs do. sorry 😢 
So every time we update `test_history` resolution, we need to update `test_resolution` as well, right?

Reply:
Yep, that's correct, but if the test the user has open, already has a `test_history_resolution`, we don't update  `test_resolution`, as there is already a specific resolution for the test on that specific run.

Probably we need to put more focus into differentiating `test_resolution` and `test_history_resolution`